### PR TITLE
 Add expand.network aggregated state price adapter

### PR DESCRIPTION
**Pull request by AtanasKrondev**
## Description

Use expand.network state data as an additional source in CL state price feeds

## Changes

- Added new adapter package  `expand-network`

## Steps to Test

Integration test, manual test

**Example request**
```json
{
  "data": {
    "endpoint": "price",
    "base": "wstETH",
    "quote": "ETH"
  }
}
```

**Example response**

```json
{
  "result": 1.1764754927001149,
  "data": {
    "result": 1.1764754927001149
  },
  "timestamps": {
    "providerDataStreamEstablishedUnixMs": 1728575700542,
    "providerDataReceivedUnixMs": 1728575720757,
    "providerIndicatedTimeUnixMs": 1728564920000
  },
  "statusCode": 200,
  "meta": {
    "adapterName": "EXPAND_NETWORK",
    "metrics": {
      "feedId": "{\"base\":\"wsteth\",\"quote\":\"eth\"}"
    }
  }
}
```

## Quality Assurance

- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `adapter-secrets` configuration file or update the [soak testing blacklist](/packages/scripts/src/get-changed-adapters/soakTestBlacklist.ts).
- [ ] If a new adapter was made, or a new endpoint was added, update the `test-payload.json` file with relevant requests.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Jira.
- [ ] This is related to a maximum of one Jira story or GitHub issue.
- [ ] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types).
- [ ] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

**Comment by mxiao-cll** [comment]
In Steps to Test please include some samples of input and output

File: packages/sources/expand-network/src/config/overrides.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (mxiao-cll) [inline packages/sources/expand-network/src/config/overrides.json:1]: Remove this file

File: packages/sources/expand-network/src/endpoint/price.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  Response: SingleNumberResultResponse
+  Settings: typeof config.settings
+}
+
+export const endpoint = new AdapterEndpoint({
+  name: 'price',
+  aliases: ['state'],
+  transport: wsTransport,
+  inputParameters,
+  overrides: overrides['expand-network'],
```
Comment (mxiao-cll) [inline packages/sources/expand-network/src/endpoint/price.ts:29]: Remove this as well

File: packages/sources/expand-network/src/endpoint/price.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export type BaseEndpointTypes = {
+  Parameters: typeof inputParameters.definition
+  Response: SingleNumberResultResponse
+  Settings: typeof config.settings
+}
+
+export const endpoint = new AdapterEndpoint({
+  name: 'price',
+  aliases: ['state'],
```
Comment (mxiao-cll) [inline packages/sources/expand-network/src/endpoint/price.ts:26]: also add crypto as an aliase

File: packages/sources/expand-network/src/transport/price.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  builders: {
+    subscribeMessage: (params) => {
+      const pair = `${params.base}/${params.quote}`.toUpperCase()
+      return {
+        action: 'liquiditymetrics',
+        asset: pair,
+      }
+    },
+    unsubscribeMessage: (params) => `UNSUBSCRIBE:${`${params.base}/${params.quote}`.toUpperCase()}`,
```
Comment (mxiao-cll) [inline packages/sources/expand-network/src/transport/price.ts:61]: Does this actually work?

**Comment by mxiao-cll** [comment]
Need to fix this failure: PR Checks / Run integration tests for changed adapters (pull_request)

File: packages/sources/expand-network/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
+import { Adapter } from '@chainlink/external-adapter-framework/adapter'
+import { config } from './config'
+import { price } from './endpoint'
+
+export const adapter = new Adapter({
```
Comment (mxiao-cll) [inline packages/sources/expand-network/src/index.ts:6]: Use AdapterEndpoint

File: packages/sources/expand-network/src/transport/price.ts (hunk 4 above)
Comment (AtanasKrondev) [reply packages/sources/expand-network/src/transport/price.ts:61]: There's no documentation how to unusbscribe from expand.networks's WS. I'll remove the unsubscribe message.

**Comment by mxiao-cll** [comment]
- In Steps to Test please include some samples of input and output
- Need to fix this failure: PR Checks / Run integration tests for changed adapters (pull_request)

**Comment by mxiao-cll** [comment]
Can you test a failure scenario as well? like a base and quote that does not exist

**Review by mxiao-cll — APPROVED** [review]

**Pull request merged**